cornell-dti/office-hours
# Adding snackbars, banners, and new twilio config to text notifications PR

### Summary 


(Note this is a merge into my other PR, not into master)   
Adds the new twilio config, generic snackbars/banners integrated with redux.


- Adds three types of Announcements: snackbars, sessionBanners, and Banners
- Replaces "enable browser notifications" and time warning/time up banners with new generic banners
- Adds snackbar for enabling/updating/disabling text notifications
- Adds "enable text notifications" banner that shows once to a user who has not seen it before






### Test Plan 


- Enable, update, and disable text notifications to test the snackbar ui works
- Reset notification settings to test the banner displays, and ensure it doesn't display if you have disabled notifications
- Make sure you see the text notification banner the first time you log in, and changed textPrompted in the database to false to see it again
- Test enabling time constraints on a course, and ensure the banners display correctly for warning and time is up. 

### Notes 



### Breaking Changes 

None
